src/app/tide/spectrum_preprocess2.cc

```diff
@@ -231,24 +184,20 @@ void ObservedPeakSet::PreprocessSpectrum(const Spectrum& spectrum, int charge,
       }
 
       // added by Yang
-      // sort region_peaks w.r.t the descending intensity
-      sort(region_peaks.begin(), region_peaks.end(), [](const pair<int, double> &left, const pair<int, double> &right) { return left.second > right.second; });
-      // save the top samanda-regional-topk peaks per region
-      for (int peak_idx=0; peak_idx<region_peaks.size(); ++peak_idx) {
-    	  if (peak_idx >= Params::GetInt("msamanda-regional-topk")) { break; }
-    	  // carp(CARP_DETAILED_DEBUG, "Region:[%d, %d] \t total_peaks: %d \t Peak mzbin: %d \t intensity: %f", i * region_size, (i+1) * region_size, region_peaks.size(), region_peaks[peak_idx].first, region_peaks[peak_idx].second );
-    	  // int peak_mzbin = region_peaks[peak_idx].first;
+      if (dia_mode) {
```

Comment:
The code line, above `region_peaks.push_back(make_pair(index, peaks_[index]));` should also be in a dia_mode block

Reply:
The issue in line 214 (or 167) is still there.

Reply:
The issue in line 214 (or 167) is still there.